public/app/features/dashboard/api/v0.ts

```diff
@@ -60,13 +63,21 @@ export class K8sDashboardAPI implements DashboardAPI<DashboardDTO> {
   }
 
   asSaveDashboardResponseDTO(v: Resource<DashboardDataDTO>): SaveDashboardResponseDTO {
+    const url = locationUtil.assureBaseUrl(
+      getDashboardUrl({
+        uid: v.metadata.name,
+        currentQueryParams: '',
+        slug: kbn.slugifyForUrl(v.spec.title),
```

Comment:
This is where we need the slug @ryantxu  - otherwise we are changing route causing the dashboard to be re-fetched unnecessarily.